Mathlib/Data/Fin/Basic.lean

```diff
@@ -537,6 +538,18 @@ protected theorem zero_add [NeZero n] (k : Fin n) : 0 + k = k := by
 instance [NeZero n] : OfNat (Fin n) a where
   ofNat := Fin.ofNat' a (NeZero.pos n)
 
+instance inhabited (n : ℕ) [NeZero n] : Inhabited (Fin n) :=
+  ⟨0⟩
+
+instance inhabitedFinOneAdd (n : ℕ) : Inhabited (Fin (1 + n)) :=
```

Comment:
This looks like `default` might be definitionally weird, I think? You could probably just prove it directly with ⟨(0, by simp)⟩ instead, I guess

Reply:
Or the way the old version does it, which does reduce correctly

Reply:
Just to be clear, are you referring to my current
```
  haveI : NeZero (1 + n) := by rw [Nat.add_comm]; infer_instance
  inferInstance
```
proof? Remember that `haveI` inlines and in this case is a prop, so this will reduce directly to the instance above.

Reply:
That's reasonable, but then isn't this change just giving a longer proof for a trivial instance?

Reply:
Yes, but this spelling makes it clear that's it's a special case of the existing instance